syz-ci/manager.go

```diff
@@ -555,6 +571,7 @@ func (mgr *Manager) uploadBuild(info *BuildInfo, imageDir string) (string, error
 	if err := mgr.dash.UploadBuild(build); err != nil {
 		return "", err
 	}
+	mgr.uploadBuildArtifacts(build, imageDir)
```

Comment:
A simpler way to do this is to upload assets first, then upload the build info to the dashboard once with all of the assets included; rather than upload assets afterwards and add them one-by-one. This way the dashboard API is simpler and we don't have half-filled objects in the datastore.
What's the advantage of doing it the current way?

Reply:
Errr... cover reports :(

We may already report a bug (in particular for broken build) but don't have build artifacts yet.

Also we may upload the same build multiple times, e.g. LTS releases may change once per 1-2 weeks, but we restart and upload build every day. Currently we will re-upload duplicates of images to GCS, right? Could we avoid that?

Reply:
Now the code only uploads build assets after a rebuild is done (unless we're in debug mode). Also, adjusted the API -- now it can take assets both per separate API call or during build creation.